TestBenches/TrackMerger_test.cpp

```diff
@@ -66,67 +67,56 @@ int main(){
-      diskStubWords_0 << track.getDiskStubWord<4>();
-      diskStubWords_1 << track.getDiskStubWord<5>();
-      diskStubWords_2 << track.getDiskStubWord<6>();
-      diskStubWords_3 << track.getDiskStubWord<7>();
+      } 
+      trackWord[i] = track.getTrackWord();
+      barrelStubWords[i][0] = track.getBarrelStubWord<0>();
+      barrelStubWords[i][1] = track.getBarrelStubWord<1>();
+      barrelStubWords[i][2] = track.getBarrelStubWord<2>();
+      barrelStubWords[i][3] = track.getBarrelStubWord<3>();
+      diskStubWords[i][0] = track.getDiskStubWord<4>();
+      diskStubWords[i][1] = track.getDiskStubWord<5>();
+      diskStubWords[i][2] = track.getDiskStubWord<6>();
+      diskStubWords[i][3] = track.getDiskStubWord<7>();
     }
 
     // Unit under test
     TrackMergerTop(bx,
     trackWord,
-    barrelStubWords_0, 
-    barrelStubWords_1,
-    barrelStubWords_2,
-    barrelStubWords_3,
-    diskStubWords_0,
-    diskStubWords_1,
-    diskStubWords_2,
-    diskStubWords_3,
+    barrelStubWords,
+    diskStubWords,
     bx_o,
     trackWord_o,
-    barrelStubWords_0_o,
-    barrelStubWords_1_o,
-    barrelStubWords_2_o,
-    barrelStubWords_3_o,
-    diskStubWords_0_o,
-    diskStubWords_1_o,
-    diskStubWords_2_o, 
-    diskStubWords_3_o
+    barrelStubWords_o,
+    diskStubWords_o
     );
 
     bool truncation = false;
 
     // Filling outputs
     unsigned nTracks = 0;
-    for (unsigned short i = 0; i < kMaxTrack+kNComparisonModules; i++){ //first 16 CMs don't appear in output first if outputting unmerged tracks first
+    for (unsigned short i = 0; i < kMaxTrack; i++){ 
       TrackFitType track;
-      track.setTrackWord(trackWord_o.read());
-      track.setBarrelStubWord<0>(barrelStubWords_0_o.read());
-      track.setBarrelStubWord<1>(barrelStubWords_1_o.read());
-      track.setBarrelStubWord<2>(barrelStubWords_2_o.read());
-      track.setBarrelStubWord<3>(barrelStubWords_3_o.read());
-      track.setDiskStubWord<4>(diskStubWords_0_o.read());
-      track.setDiskStubWord<5>(diskStubWords_1_o.read());
-      track.setDiskStubWord<6>(diskStubWords_2_o.read());
-      track.setDiskStubWord<7>(diskStubWords_3_o.read());
+      track.setTrackWord(trackWord_o[i]);
+      track.setBarrelStubWord<0>(barrelStubWords_o[i][0]);
+      track.setBarrelStubWord<1>(barrelStubWords_o[i][1]);
+      track.setBarrelStubWord<2>(barrelStubWords_o[i][2]);
+      track.setBarrelStubWord<3>(barrelStubWords_o[i][3]);
+      track.setDiskStubWord<4>(diskStubWords_o[i][0]);
+      track.setDiskStubWord<5>(diskStubWords_o[i][1]);
+      track.setDiskStubWord<6>(diskStubWords_o[i][2]);
+      track.setDiskStubWord<7>(diskStubWords_o[i][3]);
       
       outputTracks.write_mem(bx, track, nTracks );
       ++nTracks;
+      
     } 
 
     // Comparing outputs
     err_count += compareMemWithFile<TrackFitMemory<NBarrelStub, NDiskStub>>(outputTracks, fout_outputTracks.at(0), ievt, "Tracks", truncation);
-    
+    outputTracks.clear();
```

Comment:
@kwalkingshaw I think this is superfluous, since you have the same thing here:
https://github.com/cms-L1TK/firmware-hls/blob/e7b4815cb98c9303643b60dd2f91a77bd0b45040/TestBenches/TrackMerger_test.cpp#L57